src/third_party/blake3/CMakeLists.txt

```diff
@@ -17,6 +17,16 @@ include(CheckCCompilerFlag)
 
 function(add_source_if_enabled feature compile_flags)
   string(TOUPPER "have_${blake_source_type}_${feature}" have_feature)
+
+  # AVX512 support fails to compile with old Apple Clang versions even though
+  # the compiler accepts the -m flags.
+  if(${feature} STREQUAL "avx512"
+      AND CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang"
+      AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 9.0)
+    message(STATUS "Detected unsupported compiler for ${have_feature} - disabled")
+    set(${have_feature} FALSE)
```

Comment:
Will it be overwritten ~4 lines later? I do not know.

```
  if(NOT DEFINED have_feature)
    #MESSAGE(STATUS "Running automatic feature detection...")
    if(${blake_source_type} STREQUAL "asm")
      check_asm_compiler_flag(${compile_flags} ${have_feature})
    else()
      check_c_compiler_flag(${compile_flags} ${have_feature})
    endif()
  endif()
```

Reply:
The two `check_X_compiler_flag` functions doesn't do anything if the variable is already set (so that the test wont be re-run if you run cmake a second time). But I agree that it might not be obvious from the code. I'll change the code slightly.

Reply:
Ah ok, thanks. Maybe a comment will suffice. At least I did now know this about this behavior, so other will also not know it.

Reply:
Hopefully better now.